### ISSUE TITLE
Missing "Navigation" landmark on NavigationView when pane toggle button is hidden

### ISSUE THREAD

**Issue by vgromfeld** [microsoft/microsoft-ui-xaml#7781]
### Describe the bug

There is no `NavigationLandmark` when the `NavigationView`'s pane toggle button is hidden ([NavigationView.IsPaneToggleButtonVisible](https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.navigationview.ispanetogglebuttonvisible?view=winui-2.8) is `false`).

### Steps to reproduce the bug

1. Add the following code in an app
```xml
<mux:NavigationView IsPaneToggleButtonVisible="False" DisplayMode="Left">
    <mux:NavigationView.MenuItems>
        <mux:NavigationViewItem Content="Item1" />
        <mux:NavigationViewItem Content="Item2" />
    </mux:NavigationView.MenuItems>
</mux:NavigationView>
```
2. Run the app
3. Start Windows' narrator using CTRL+WIN+ENTER
4. Activate landmarks window using CAPS LOCK+F5
5. Check the list of available landmarks
6. There is no navigation landmark to focus.

### Expected behavior

The left nav area must have a navigation landmark to help users with assistive technology to find the main navigation area.

### NuGet package version

Microsoft.UI.Xaml 2.8.0

### Windows app type

- [X] UWP
- [ ] Win32

### Windows version

Windows 11 (22H2): Build 22621

### Additional context

The issue comes from [NavigationView.xaml#L238](https://github.com/microsoft/microsoft-ui-xaml/blob/8ac2866a2b42165184a265181caa931f6638ccbd/dev/NavigationView/NavigationView.xaml#L238).
The landmark is only defined on the pane toggle button which can be hidden.
